### PR TITLE
[DOCS] Document the environment variable substitution feature of kustomize configMapGenerator

### PR DESCRIPTION
**Pull request by afirth**
Document the environment variable substitution feature of the kustomize configMapGenerator

/Fixes https://github.com/kubernetes-sigs/cli-experimental/issues/66

Relates:
https://github.com/kubernetes-sigs/kustomize/issues/2497

Code with comment docs:
https://github.com/kubernetes-sigs/kustomize/blob/master/api/kv/kv.go#L136
https://github.com/kubernetes-sigs/kustomize/blob/master/api/kv/kv.go#L165

### CODE REVIEW

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md
Comment (geoffcline) [inline content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: can you make these names more descriptive? I'd like to avoid "foo", "bar", and "baz". Overall I think this change is worthwhile :)

**Review by geoffcline — CHANGES_REQUESTED** [review]

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md
Comment (afirth) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: Do you have a concrete suggestion for better names? I was following the convention before me - I would need to update the names in the other sections which I am happy to do if you have ideas, but probably better in a separate PR

Comment (afirth) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: @geoffcline I just checked and there are 10 instances of `FOO` in the file. The editor would need to carefully check the output matches the new inputs. I feel this is out of scope of this PR.

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md
Comment (RinkiyaKeDad) [inline content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:128]: nit: do we need to change this too?

Comment (afirth) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:128]: This is the encoded SHA256 of the configmap [generated by kustomize](https://github.com/kubernetes/cli-runtime/blob/v0.20.12/pkg/kustomize/k8sdeps/transformer/hash/hash.go#L162). So yes, I believe it should change to reflect the output a user running the examples would see.

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md
Comment (geoffcline) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: I've thought about this. Unfortunately, I don't feel comfortable approving something that includes lines explicitly prohibited by the style guide https://kubernetes.io/docs/contribute/style/style-guide/#inline-code-formatting

You can ask in slack? or just try to remove "foo" in the examples you are updating?

thanks, Geoffrey

Comment (afirth) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: Hi @geoffcline,

Firstly, thanks for taking the time to review.

I would strongly argue that the use of `foo` and `bar` is entirely appropriate here. They are explicitly the standard names for something that is "not meaningful and lack[s] context" because any meaning and context would detract from the example. The analogous example would be using `lorem ipsum` in a template example. Anything else is either a) a less common substitute for `lorem ipsum` or b) misleading.

So I see a few ways forward to fixing this missing documentation:
1) suggest better variable names and I will implement them
2) PR the style guide for an exception
  a) I don't love this, because the style guide should be simple and cover the common case - outside of tests and examples of arbitrary variable handling there are not many cases where these would be appropriate names
3) accept this PR to fix the [open issue](https://github.com/kubernetes-sigs/cli-experimental/issues/66) and open a separate issue to build consensus on 1) or 2)
4) move to another channel or tag another reviewer / approver to build consensus
5) accept my argument and "lgtm" :smile: 

TBH I find this pretty hard, I'm contributing docs for something that annoys and even blocks a lot of people in my free time and IMO it feels like an overall improvement being blocked on a nit with no clear suggestion on how to resolve the nit. The contribution matches the existing code style. I can't just remove the `FOO` - the example becomes meaningless.

Thanks again for your time and assistance to resolve
-Aalstair

**Comment by afirth** [comment]
thanks @jimangel! According to the bot, you are also the chosen approver?
/assign @jimangel

File: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md
Comment (afirth) [reply content/en/docs/tasks/manage-kubernetes-objects/kustomization.md:103]: resolving due to no path to resolution and inactivity + separate /lgtm from @jimangel

**Comment by afirth** [comment]
@jimangel rebased and awaiting lgtm + approve

**Review by onlydole — APPROVED** [review]
Thank you @afirth! This looks good to me.

**Comment by afirth** [comment]
Thanks @onlydole @jimangel , just need another /lgtm as the rebase revoked it

**Comment by annajung** [comment]
https://deploy-preview-30348--kubernetes-io-main-staging.netlify.app/docs/tasks/manage-kubernetes-objects/kustomization/

/lgtm

**Pull request merged**
